src/types/coefficient.rs

```diff
@@ -141,6 +141,22 @@ impl TryFrom<Coefficient> for BigInt {
     }
 }
 
+impl TryFrom<BigInt> for Coefficient {
+    type Error = IonError;
+
+    fn try_from(value: BigInt) -> Result<Self, Self::Error> {
+        let (sign, magnitude) = value.into_parts();
+        let sign = match sign {
+            num_bigint::Sign::Minus => Sign::Negative,
+            num_bigint::Sign::Plus => Sign::Positive,
+            num_bigint::Sign::NoSign => {
+                return illegal_operation("Cannot convert sign-less BigInt to Decimal.")
```

Comment:
I was really hoping [the documentation for `NoSign`](https://docs.rs/num-bigint/latest/num_bigint/enum.Sign.html#variants) would shed some light on its intended meaning, but no dice. :/

What does it mean for an integer to be signless?

Reply:
I had hoped the same thing. 

Actually, looking through the source, the implication is that `NoSign` should only be valid for `0` ?

Cf.
- https://github.com/rust-num/num-bigint/blob/6f2b8e0fc218dbd0f49bebb8db2d1a771fe6bafa/src/bigint/division.rs#L51
- https://github.com/rust-num/num-bigint/blob/6f2b8e0fc218dbd0f49bebb8db2d1a771fe6bafa/src/bigrand.rs#L93-L96
- https://github.com/rust-num/num-bigint/blob/6f2b8e0fc218dbd0f49bebb8db2d1a771fe6bafa/src/bigint/shift.rs#L86C1-L88

Reply:
Cf. https://github.com/rust-num/num-bigint/blob/6f2b8e0fc218dbd0f49bebb8db2d1a771fe6bafa/src/bigint.rs#L211-L230